extension/icu/icu-timebucket.cpp

```diff
@@ -0,0 +1,725 @@
+#include "duckdb/common/exception.hpp"
+#include "duckdb/common/limits.hpp"
+#include "duckdb/common/operator/cast_operators.hpp"
+#include "duckdb/common/types/interval.hpp"
+#include "duckdb/common/types/time.hpp"
+#include "duckdb/common/types/timestamp.hpp"
+#include "duckdb/common/types/value.hpp"
+#include "duckdb/common/vector_operations/binary_executor.hpp"
+#include "duckdb/common/vector_operations/ternary_executor.hpp"
+#include "duckdb/parser/parsed_data/create_scalar_function_info.hpp"
+#include "include/icu-datefunc.hpp"
+#include "include/icu-timezone.hpp"
+
+namespace duckdb {
+
+struct ICUFromLocalTime : public ICUDateFunc {
```

Comment:
Currently I use `ICUFromNariveTimestamp` only in unavoidable cases such as origin conversion.

Reply:
I will give some thought to if this implementation will cause any problems.

Reply:
In my understanding, the conversion from `TIMESTAMP` to `TIMESTAMPTZ` and vice versa have no ambiguity problem. The ambiguity can appear during binning, but it is avoided because operations are performed by `ICUDateFunc` (or by primitive types if it can be converted to a single value of either microseconds, days, or months to remove ambiguity).